apps/react/src/features/studios/components/StudioLayout/StudioLayout.tsx

```diff
@@ -0,0 +1,79 @@
+import { memo, FC, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { selectIsDrawerOpen } from '@js-camp/react/store/drawer/selectors';
+import { useAppDispatch, useAppSelector } from '@js-camp/react/store';
+import {
+	selectCursor, selectHasMoreData, selectIsPaginationEvent, selectStudios,
+} from '@js-camp/react/store/studio/selectors';
+import { Box, ListItemText, IconButton } from '@mui/material';
+import AddIcon from '@mui/icons-material/Add';
+import { useNavigate, Outlet, useParams } from 'react-router-dom';
+import { fetchStudios } from '@js-camp/react/store/studio/dispatchers';
+import { StudioQueryParameters } from '@js-camp/core/models/studio-query-parameters.model';
+
+import { useQueryParams } from '../../hooks/useQueryParams';
+
+import { StudiosList } from '../StudioList';
+
+import styles from './StudioLayout.module.css';
+
+const StudioLayoutComponent: FC = () => {
+	const open = useSelector(selectIsDrawerOpen);
+	const navigate = useNavigate();
+	const { studioId } = useParams<{ studioId: string; }>();
+	const dispatch = useAppDispatch();
+	const studios = useAppSelector(selectStudios);
+	const cursor = useAppSelector(selectCursor) ?? undefined;
+	const hasMoreData = useAppSelector(selectHasMoreData);
+	const isPaginationEvent = useAppSelector(selectIsPaginationEvent);
+
+	const { queryParams, setQueryParams } = useQueryParams();
+
+	const params: StudioQueryParameters = {
+		ordering: queryParams.get('ordering') ?? undefined,
+		search: queryParams.get('search') ?? undefined,
+		cursor: cursor ?? queryParams.get('cursor') ?? undefined,
+	};
+
+	useEffect(() => {
+		setQueryParams({
+			...queryParams,
+			ordering: queryParams.get('ordering') ?? undefined,
+			search: queryParams.get('search') ?? undefined,
+			cursor: cursor ?? queryParams.get('cursor') ?? undefined,
+		});
+	}, [cursor, queryParams]);
+
+	useEffect(() => {
+		if (!isPaginationEvent || isPaginationEvent && hasMoreData) {
+			dispatch(fetchStudios(params));
+		}
+	}, [dispatch, queryParams]);
+
+	const handleStudioClick = (id: number) => {
+		navigate(`/studio/${id}`);
+	};
+
+	return (
+		<main className={`${styles.layout} ${open ? styles.layout_open : ''}`}>
```

Comment:
Please use `clsx` for these cases to make code more readable. https://www.npmjs.com/package/clsx
```suggestion
		<main className={clsx(
		        styles.layout,
		        open && styles.layout_open
		 )}>
```